src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -23,24 +25,38 @@ public class ModelManager implements Model {
     private final AddressBook addressBook;
     private final UserPrefs userPrefs;
     private final FilteredList<Person> filteredPersons;
+    private final FilteredList<Item> filteredItems;
+    private final Catalogue catalogue;
+
 
     /**
      * Initializes a ModelManager with the given addressBook and userPrefs.
      */
-    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs) {
-        requireAllNonNull(addressBook, userPrefs);
+    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs, Catalogue catalogue) {
```

Comment:
I was thinking the catalogue was part of the addressBook, but seems like here it was implemented as part of a model manager? If ModelManager requires the itemlist, it can use just the getItemList() method in ReadOnlyAddressBook. This change in ModelManager results in many changes in all the test cases, maybe can see if theres a way to avoid this change?